lib/active_merchant/billing/gateways/garanti.rb

```diff
@@ -230,7 +230,7 @@ def commit(money,request)
       end
 
       def parse(body)
-        xml = REXML::Document.new(body)
+        xml = REXML::Document.new(strip_invalid_xml_chars(body))
```

Comment:
Why rebuild the REXML doc if it's being done in the method? Maybe the strip method should just run the gsub no matter what?